### PR TITLE
Refactor Lesson 04

### PR DESCRIPTION
**Pull request by billmwong**
Made a separate Lecture.md file so the stuff I do in lecture doesn't clutter up the readme, and changed some minor wording.

The rest of the readme is basically just a walkthrough of the formsApp, which builds upon itself and teaches as it makes you build the formsApp.  This doesn't really make sense for a learn-share.

### CODE REVIEW

File: lessons/04-client-jquery-ajax/README.md [hunk 1]
```diff
@@ -1,113 +1,35 @@
...
 
-![Cookie Clicker](images/cookie1.png)
+Today we are going to learn the basics of building and manipulating the client and building communication routes between the client and server.
 
-If you investigate (or just guess) you will notice a variable called "Game" and if you start to type that into the console it will even suggest an autocomplete for it. You can look at its functions and realize that there is a variable called cookiePs, which you might guess corresponds to cookies per second. You can set cookiesPs to any value and see what happens:
-
-![Cookie Clicker hacked](images/cookie2.png)
-
-The takeaway from all this is that the client is NOT secure. A smart user can take your app apart on the client and do anything he wants with it. All security needs to be implemented on the server side so that it cannot be meddled with. 
+## README Table of Contents
```
Comment (yunhsincynthiachen) [inline lessons/04-client-jquery-ajax/README.md]: Can we have the table of contents also include a link to the in-class and the homework?

File: lessons/04-client-jquery-ajax/README.md [hunk 2]
```diff
@@ -486,38 +330,21 @@ $form.submit(function(event) {
...
 
 ##Additional and Alternative learning Resources
 A free Google-sponsered course on the devtools: https://www.codeschool.com/courses/discover-devtools
 
-#Lesson 5 In-Class Exercises
-
-##Class 5 README
-Take some time to finish up your forms app or ask any questions if you need to, and push your solution to your fork. Our solution to the walkthrough is [here](https://github.com/olinjs/olinjs/blob/master/lessons/04-client-jquery-ajax/formsAppWalkthrough). 
+#Lesson 4 In-Class Exercises
+Follow along above to finish up your forms app and ask any questions if you need to. Our solution to the walkthrough is [here](./formsAppWalkthrough).
```
Comment (yunhsincynthiachen) [inline lessons/04-client-jquery-ajax/README.md]: Weren't we going to get rid of the itsChristmas refactor?

File: lessons/04-client-jquery-ajax/README.md [hunk 3]
```diff
@@ -554,52 +381,16 @@ Last class we discussed folder structure, so as a refresher, draw out the folder
...
-
-#### Preclass Reading and Exercise
-- Read the [Class 6 README](https://github.com/olinjs/olinjs/tree/master/lessons/06-css-development-style) before class 6 on Friday 2/5/16.
-- Send an email to [olinjs16@gmail.com](olinjs16@gmail.com) with the subject line "Preclass 6" telling us about...
-    - Something in the reading you felt confident about and easily grasped
-    - Something in the reading you're confused about or want to know more about
+  * Out-of-Stock button will tell the server to label the ingredient as disabled. The ingredient should be removed from the current page without refreshing. (Optional: make it toggleable to "add" more of the ingredient. In this case, do not remove the ingredient from the page, but make note through words or style that it is unavailable.)
+  * Edit button allows the user to submit a new name or price for the ingredient which the server will update. The edits should change the ingredient list without refreshing.
+
+When you're finished, fill out the [Homework 4 Submission Survey] ().
```
Comment (yunhsincynthiachen) [inline lessons/04-client-jquery-ajax/README.md]: Here's the Homework 4 Submission survey: https://docs.google.com/forms/d/e/1FAIpQLSd-ths9fq5m28ooRoiDwiYjRc6s48JBZDgCmFQqNEGUqOb6hw/viewform

File: lessons/04-client-jquery-ajax/README.md [hunk 4]
```diff
@@ -554,52 +381,16 @@ Last class we discussed folder structure, so as a refresher, draw out the folder
...
-When you're finished, fill out the [Homework 5 Submission Survey] (https://docs.google.com/forms/d/1MsGf_gNkHJBjBEzQoaUXvmpc3Nljk838FeBXlJ5hH9c/viewform?usp=send_form). 
-
-#### Preclass Reading and Exercise
-- Read the [Class 6 README](https://github.com/olinjs/olinjs/tree/master/lessons/06-css-development-style) before class 6 on Friday 2/5/16.
-- Send an email to [olinjs16@gmail.com](olinjs16@gmail.com) with the subject line "Preclass 6" telling us about...
-    - Something in the reading you felt confident about and easily grasped
-    - Something in the reading you're confused about or want to know more about
+  * Out-of-Stock button will tell the server to label the ingredient as disabled. The ingredient should be removed from the current page without refreshing. (Optional: make it toggleable to "add" more of the ingredient. In this case, do not remove the ingredient from the page, but make note through words or style that it is unavailable.)
+  * Edit button allows the user to submit a new name or price for the ingredient which the server will update. The edits should change the ingredient list without refreshing.
+
```
Comment (yunhsincynthiachen) [inline lessons/04-client-jquery-ajax/README.md:384]: Can we say something here like: This homework is the first part of a bigger application. If you'd like to see the other endpoints that the application will need, look [here] ().

File: lessons/04-client-jquery-ajax/README.md [hunk 5]
```diff
@@ -358,39 +361,27 @@ Last class we discussed folder structure, so as a refresher, draw out the folder
...
+For Friday, your application will need the following http endpoint:
 * `/ingredients` =>
   * Shows a list of current ingredients (Name and Price) with Out-of-Stock and edit button.
   * An `Add` button should allow the user to specify the name and price of a new ingredient which should appear on the page without requiring a refresh.
   * Out-of-Stock button will tell the server to label the ingredient as disabled. The ingredient should be removed from the current page without refreshing. (Optional: make it toggleable to "add" more of the ingredient. In this case, do not remove the ingredient from the page, but make note through words or style that it is unavailable.)
   * Edit button allows the user to submit a new name or price for the ingredient which the server will update. The edits should change the ingredient list without refreshing.
 
-When you're finished, fill out the [Homework 4 Submission Survey] (). 
\ No newline at end of file
+The next homework that you will be doing will build off of this homework. If you'd like to see the other endpoints that the application will need, look [here] (../05-flex-burger-help/README.md#L58).
```
Comment (billmwong) [inline lessons/04-client-jquery-ajax/README.md]: It's not called README right now, it's called INCLASS. I think it should be called readme though, can you rename it in this branch?

Comment (yunhsincynthiachen) [reply lessons/04-client-jquery-ajax/README.md]: I did. The readme in 05-flex-burger-help is called README, not INCLASS anymore.

Comment (billmwong) [reply lessons/04-client-jquery-ajax/README.md]: Oh you're right, didn't see that

**Comment by billmwong** [comment]
lgtm :ship:

**Pull request merged**